internal/templateutil/state.go

```diff
@@ -160,6 +146,33 @@ func ParseAccessRules(ctx context.Context, cl client.Client, rules []hmc.AccessR
 	return expectedState, nil
 }
 
+func IsAvailableForUpgrade(ctx context.Context, cl client.Client, templateKind hmc.TemplateKind, source, target string) (bool, error) {
+	tmList := &hmc.TemplateManagementList{}
+	err := cl.List(ctx, tmList)
+	if err != nil {
+		return false, err
+	}
+	if len(tmList.Items) == 0 {
+		return false, fmt.Errorf("TemplateManagement is not found")
+	}
+	for _, accessRule := range tmList.Items[0].Spec.AccessRules {
```

Comment:
I am not in the context, why only the first entry is being looped over? Would it be a good idea to comment on the idea behind it?